tests/components/wled/test_light.py

```diff
@@ -79,82 +79,96 @@
+            LIGHT_DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: "light.wled_rgb_light", ATTR_TRANSITION: 5},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        light_mock.assert_called_once_with(
+            on=False, segment_id=0, transition=50,
+        )
+
+    with patch("wled.WLED.light") as light_mock:
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_ON,
+            {
+                ATTR_BRIGHTNESS: 42,
+                ATTR_EFFECT: "Chase",
+                ATTR_ENTITY_ID: "light.wled_rgb_light",
+                ATTR_RGB_COLOR: [255, 0, 0],
+                ATTR_TRANSITION: 5,
+            },
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        light_mock.assert_called_once_with(
+            brightness=42,
+            color_primary=(255, 0, 0),
+            effect="Chase",
+            on=True,
+            segment_id=0,
+            transition=50,
+        )
+
+    with patch("wled.WLED.light") as light_mock:
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "light.wled_rgb_light", ATTR_COLOR_TEMP: 400},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        light_mock.assert_called_once_with(
+            color_primary=(255, 159, 70), on=True, segment_id=0,
+        )
 
-    await hass.services.async_call(
-        LIGHT_DOMAIN,
-        SERVICE_TURN_ON,
-        {
-            ATTR_BRIGHTNESS: 42,
-            ATTR_EFFECT: "Chase",
-            ATTR_ENTITY_ID: "light.wled_rgb_light",
-            ATTR_RGB_COLOR: [255, 0, 0],
-            ATTR_TRANSITION: 5,
-        },
-        blocking=True,
-    )
-    await hass.async_block_till_done()
 
-    state = hass.states.get("light.wled_rgb_light")
-    assert state.state == STATE_ON
-    assert state.attributes.get(ATTR_BRIGHTNESS) == 42
-    assert state.attributes.get(ATTR_EFFECT) == "Chase"
-    assert state.attributes.get(ATTR_HS_COLOR) == (0.0, 100.0)
+async def test_light_error(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, caplog
+) -> None:
+    """Test error handling of the WLED lights."""
+    aioclient_mock.post("http://example.local:80/json/state", text="", status=400)
+    entry = await init_integration(hass, aioclient_mock)
+    coordinator = hass.data[DOMAIN][entry.entry_id]
 
     await hass.services.async_call(
         LIGHT_DOMAIN,
-        SERVICE_TURN_ON,
-        {ATTR_ENTITY_ID: "light.wled_rgb_light", ATTR_COLOR_TEMP: 400},
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "light.wled_rgb_light"},
         blocking=True,
     )
     await hass.async_block_till_done()
-    state = hass.states.get("light.wled_rgb_light")
-    assert state.state == STATE_ON
-    assert state.attributes.get(ATTR_HS_COLOR) == (28.874, 72.522)
 
+    assert coordinator.last_update_success
```

Comment:
Don't assert anything about the coordinator. That's just a detail.

We want to assert what the state of the entity is after the error.

Reply:
Done!